src/main/java/frc/team5115/Classes/Software/AutoAimAndRange.java

```diff
@@ -0,0 +1,132 @@
+package frc.team5115.Classes.Software; 
+
+import frc.team5115.Constants;
+import frc.team5115.Classes.Hardware.*;
+import frc.team5115.Classes.Software.PhotonVision;
+import org.photonvision.PhotonPoseEstimator;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.estimator.PoseEstimator;
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team5115.Classes.Software.Drivetrain;
+
+import frc.team5115.Constants.*;
+import java.util.Optional;
+
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonUtils;
+
+
+
+public class AutoAimAndRange extends SubsystemBase{
+    HardwareDrivetrain j;
+    Drivetrain d;
+    NAVx gyro;
+    PhotonVision photonVision;
+    int x = 0;
+
+    
+    public AutoAimAndRange(HardwareDrivetrain j, PhotonVision photonVision){  
+        this.j = j;
+        this.photonVision = photonVision;
+        x=1;
+
+
+    }
+
+
+    PhotonCamera camera = new PhotonCamera("Microsoft_LifeCam_HD-3000");
+
+    // PID constants should be tuned per robot
+    final double LINEAR_P = 0.03;
+    final double LINEAR_D = 0.0;
+    PIDController forwardController = new PIDController(LINEAR_P, 0, LINEAR_D);
+
+    final double ANGULAR_P = 0.0025;
+    final double ANGULAR_D = 0.0;
+    PIDController turnController = new PIDController(ANGULAR_P, 0, ANGULAR_D);
+
+
+
+
+// Calculate robot's field relative pose
+
+    public double[] periodic1() { 
```

Comment:
Suggestion, rename this function.

`public double[] calculateFieldRelativePose()` 

is more descriptive that `periodic1()` and makes it clear that it returns a value.